src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/XdocUtil.java

```diff
@@ -69,6 +69,23 @@ public static Set<Path> getXdocsFilePaths() throws IOException {
         }
     }
 
+    /**
+     * Gets xdocs template file paths. These are files ending with .xml.template.
+     *
+     * @return a set of xdocs template file paths.
+     * @throws IOException if an I/O error occurs.
+     */
+    public static Set<Path> getXdocsTemplatesFilePaths() throws IOException {
```

Comment:
Please provide a reason this method needs to be here and not in `XdocGenerator`.

If `XdocGenerator` is one of the classes that will be removed, then this method should go with it since we won't have template extensions in the final product unless I misunderstood something.

Reply:
I didn't have a reason, but because of https://github.com/checkstyle/checkstyle/issues/13428 we will need it.

`<macro>`s are present it `.tempate` files only.

I am adding an `until https://github.com/checkstyle/checkstyle/issues/13426` on this method.

Reply:
Ok, we could still move this to `XdocGenerator` but I am ok.